src/commands/deployWorkspaceProject/DeployWorkspaceProjectSettings.ts

```diff
@@ -0,0 +1,37 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { WorkspaceFolder } from "vscode";
+import { settingUtils } from "../../utils/settingUtils";
+
+export interface DeployWorkspaceProjectSettings {
+    // Container app names are unique to a resource group
+    containerAppResourceGroupName?: string;
+    containerAppName?: string;
+
+    containerRegistryName?: string;
+}
+
+const deployWorkspaceProjectPrefix: string = 'deployWorkspaceProject';
+
+export async function getDeployWorkspaceProjectSettings(rootFolder: WorkspaceFolder): Promise<DeployWorkspaceProjectSettings | undefined> {
+    const settingsPath: string = settingUtils.getDefaultRootWorkspaceSettingsPath(rootFolder);
+
+    try {
+        const containerAppName: string | undefined = settingUtils.getWorkspaceSetting(`${deployWorkspaceProjectPrefix}.containerAppName`, settingsPath);
+        const containerAppResourceGroupName: string | undefined = settingUtils.getWorkspaceSetting(`${deployWorkspaceProjectPrefix}.containerAppResourceGroupName`, settingsPath);
+        const containerRegistryName: string | undefined = settingUtils.getWorkspaceSetting(`${deployWorkspaceProjectPrefix}.containerRegistryName`, settingsPath);
+
+        if (containerAppName || containerAppResourceGroupName || containerRegistryName) {
+            return {
+                containerAppName,
+                containerAppResourceGroupName,
+                containerRegistryName
+            };
+        }
+    } catch { /** Do nothing */ }
```

Comment:
Might be useful to add a comment here that explains what errors are being ignored here and why

Reply:
Ok now that I see this again, you should probably log something inside that catch block too.

Reply:
This was a good suggestions, because it actually pushed me to test the get setting method a little bit more, and I realized that it doesn't actually throw an error even with bad settings path and property... so I removed the catch entirely.  I also made a small change to the setting utils so that it reads a missing value as undefined rather than as an empty string `''`.

Reply:
Ok that's great